lib/ansible/modules/packaging/os/satellite_subscription.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2019, Luc Stroobant (luc.stroobant@wdc.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+
+DOCUMENTATION = """
+---
+module: satellite_subscription
+short_description: Manage host subscriptions (licenses) on Red Hat Satellite
+description:
+  - This module allows to assign or remove subscriptions to hosts in Red Hat Satellite.
+  - It operates using the Red Hat Satellite REST API, rather than make changes directly to target systems like M(redhat_subscription).
+    Your hosts or hypervisors should be in the Satellite already. The module will run subscription
+    related tasks (assign, remove, autoattach or replace with hypervisor subscriptions) for the
+    hosts through the Satellite API.
+version_added: "2.8"
+requirements:
+  - requests
+author:
+  - Luc Stroobant (@stroobl)
+options:
+  satellite_hostname:
+    description:
+      - The name or IP address of the Red Hat Satellite server.
```

Comment:
Please add the type to every parameter in the documentation. The newer documentation generates the parameter types as part of the parameter listing and is very useful.
```suggestion
      - The name or IP address of the Red Hat Satellite server.
      type: str
```